pkgs/top-level/splice.nix

```diff
@@ -0,0 +1,76 @@
+# The `splicedPackages' package set, and its use by `callPackage`
+#
+# The `buildPackages` pkg set is a new concept, and the vast majority package
+# expression (the other *.nix files) are not designed with it in mind. This
+# presents us with a problem with how to get the right version (build-time vs
+# run-time) of a package to a consumer that isn't used to thinking so cleverly.
```

Comment:
The splicing is slow. I could imagine a `double` `callPackage` which takes a `{ .. } @ buildOrNative: { .. } @ foreignOrRun: ..`